sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdChannelAcquisitionTimeline.java

```diff
@@ -39,20 +44,35 @@ public static RntbdChannelAcquisitionEvent startNewEvent(
     public static RntbdPollChannelEvent startNewPollEvent(
         RntbdChannelAcquisitionTimeline timeline,
         int availableChannels,
-        int acquiredChannels) {
+        int acquiredChannels,
+        ClientTelemetry clientTelemetry) {
 
         if (timeline != null) {
             RntbdPollChannelEvent newEvent = new RntbdPollChannelEvent(availableChannels, acquiredChannels, Instant.now());
-            timeline.addNewEvent(newEvent);
+            timeline.addNewEvent(newEvent, clientTelemetry);
             return newEvent;
         }
 
         return null;
     }
 
-    private void addNewEvent(RntbdChannelAcquisitionEvent event) {
+    private void addNewEvent(RntbdChannelAcquisitionEvent event, ClientTelemetry clientTelemetry) {
         if (this.currentEvent != null) {
             this.currentEvent.complete(event.getCreatedTime());
+            if (event.getEventType().equals(RntbdChannelAcquisitionEventType.ATTEMPT_TO_CREATE_NEW_CHANNEL_COMPLETE)) {
+                ReportPayload reportPayload = new ReportPayload(ClientTelemetry.TCP_NEW_CHANNEL_LATENCY_NAME,
+                    ClientTelemetry.TCP_NEW_CHANNEL_LATENCY_UNIT);
+                ConcurrentDoubleHistogram newChannelLatencyHistogram =
```

Comment:
maybe use concurrentHashMap.computeIfAbsent to prevent race condition?

Reply:
Object in map is ConcurrentDoubleHistogram so there wont be race condition , and not sure how ,much effect it will put on perf , current way is tried and tested rigorously in the current implementation of CT . See this comment where containsKey method was also causing perf  https://github.com/Azure/azure-sdk-for-java/pull/16822#issuecomment-721344983

Reply:
As mentioned earlier race condition is handled by the value of Map which is ConcurrentDoubleHistogram . This code is getting used for more than year, i fear we might introduced perf if we change the structure of ClientTelemetryInfo variables to have ConcurrentHashMap as it come with cost, we can take this as separate work item may be ?
However i noticed we should wrap this with only when CT is enable , like we did in regular request latency [here](https://github.com/Azure/azure-sdk-for-java/blob/ff077373038ff09cf967ddab3d3cf4f0649ca60a/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/TracerProvider.java#L325)

Reply:
Discussed offline with Kushagra , we already have concurrent map , and we agree to test out computeIfAbsent in separate work item covering all the places where we used CT. https://github.com/Azure/azure-sdk-for-java/issues/26838 .
Also I added check to collect only when CT is enabled, I don't need to do extra perf as existing perf result i did with CT false and still this code was executing executing .